### PR TITLE
[Toolchain] Add ToolchainProvider test cases

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit adds ToolchainProvider test cases.

- Add MockCompilerWithMultipleInstalledToolchains compiler
- Add MockCompilerWithNoInstalledToolchain compiler

ONE-vscode-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

Draft PR: #1148

### CODE REVIEW

File: src/Toolchain/ToolchainProvider.ts [hunk 1]
```diff
@@ -109,138 +109,164 @@ export class ToolchainProvider implements vscode.TreeDataProvider<BaseNode> {
...
 
-  error(msg: string, ...args: string[]): Thenable<string|undefined> {
+  /* istanbul ignore next */
+  private error(msg: string, ...args: string[]): Thenable<string|undefined> {
     Logger.error(this.tag, msg);
     return vscode.window.showErrorMessage(msg, ...args);
   }
 
-  refresh() {
+  public refresh() {
```
Comment (llFreetimell) [inline src/Toolchain/ToolchainProvider.ts:130]: Maybe this can be ignored too?

File: src/Toolchain/ToolchainProvider.ts [hunk 2]
```diff
@@ -109,138 +109,164 @@ export class ToolchainProvider implements vscode.TreeDataProvider<BaseNode> {
...
+          }
+        });
   }
 
-  uninstall(tnode: ToolchainNode) {
-    if (tnode === undefined) {
-      throw Error('Invalid toolchain node.');
-    }
+  public install() {
+    /* istanbul ignore next */
```
Comment (llFreetimell) [inline src/Toolchain/ToolchainProvider.ts:193]:
```suggestion
  /* istanbul ignore next */
  public install() {
```

File: src/Toolchain/ToolchainProvider.ts [hunk 3]
```diff
@@ -109,138 +109,164 @@ export class ToolchainProvider implements vscode.TreeDataProvider<BaseNode> {
...
+    showInstallQuickInput().then(([toolchainEnv, toolchain]) => {
+      this._install(toolchainEnv, toolchain);
+    }, () => this._notifyInstallationCancelled());
+  }
+
+  public uninstall(tnode: ToolchainNode): boolean|undefined {
+    /* istanbul ignore next */
```
Comment (llFreetimell) [inline src/Toolchain/ToolchainProvider.ts:200]:
```suggestion
  /* istanbul ignore next */
  public uninstall(tnode: ToolchainNode): boolean|undefined {
```

File: src/Toolchain/ToolchainProvider.ts (hunk 1 above)
Comment (jyoungyun) [reply src/Toolchain/ToolchainProvider.ts:149]: Could you explain your intention a little more?

Comment (llFreetimell) [reply src/Toolchain/ToolchainProvider.ts:149]: I thought that this function is about UI but my misunderstanding, sorry T.T

**Review by llFreetimell — DISMISSED** [review]
LGTM, great!!

File: src/Tests/Toolchain/ToolchainProvider.test.ts [hunk 4]
```diff
@@ -99,16 +66,30 @@ suite('Toolchain', function() {
...
         const collapsibleState = vscode.TreeItemCollapsibleState.None;
-        let node = new ToolchainNode(label, backend, toolchain);
+        let node = new ToolchainNode(label, backendName, toolchain);
+        assert.strictEqual(node.label, label);
+        assert.strictEqual(node.collapsibleState, collapsibleState);
+        assert.strictEqual(node.backendName, backendName);
+      });
+    });
+
+    suite('#constructor()', function() {
```
Comment (yunjayh) [inline src/Tests/Toolchain/ToolchainProvider.test.ts:81]:
```suggestion
```

Suite `#constructor()` is defined multiple times. How about to merge them for consistency? (And there are other duplicated suites. e.g. `createToolchainNodes()`)

**Review by llFreetimell — APPROVED** [review]
LGTM

**Review by yunjayh — APPROVED** [review]
LGTM

**Pull request merged**